app/models/project_mediaflux.rb

```diff
@@ -88,30 +87,10 @@ def self.document(project:, xml_namespace: nil)
     Nokogiri::XML.parse(xml_body)
   end
 
-  def self.create_root_ns(session_id:)
-    root_namespace = Mediaflux::Connection.root_namespace
-    namespace_request = Mediaflux::NamespaceDescribeRequest.new(path: root_namespace, session_token: session_id)
-    if namespace_request.exists?
-      Rails.logger.info "Root namespace #{root_namespace} already exists"
-    else
-      Rails.logger.info "Created root namespace #{root_namespace}"
-      namespace_request = Mediaflux::NamespaceCreateRequest.new(namespace: root_namespace, description: "TigerData client app root namespace",
-                                                                      store: Store.all(session_id: session_id).first.name,
-                                                                      session_token: session_id)
-      namespace_request.response_body
-    end
-  end
-
-  def self.prepare_parent_collection(project_parent:, session_id:)
-    get_parent = Mediaflux::AssetMetadataRequest.new(session_token: session_id, id: project_parent)
-    if get_parent.error?
-      if project_parent.include?("path=")
-        create_parent_request = Mediaflux::AssetCreateRequest.new(session_token: session_id, namespace: Mediaflux::Connection.root_collection_namespace,
-                                                                        name: Mediaflux::Connection.root_collection_name)
-        raise "Can not create parent collection: #{create_parent_request.response_error}" if create_parent_request.error?
-      else
-        raise "Error finding parent collection (#{project_parent}) #{get_parent.response_error}"
-      end
-    end
+  def self.create_root_tree(session_id:)
+    root_ns = Rails.configuration.mediaflux["api_root_collection_namespace"]
+    parent_collection = Rails.configuration.mediaflux["api_root_collection_name"]
+    req = Mediaflux::RootCollectionAsset.new(session_token: session_id, root_ns:, parent_collection:)
```

Comment:
This class has the logic to make sure the root and the required nodes below it exist (i.e. parent_collection and parent_namespace)